### ISSUE TITLE
default_service wrongly required in google_compute_region_url_map in path_matcher bock

### ISSUE THREAD

**Issue by Gilles95** [hashicorp/terraform-provider-google#7951]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.13.4
### Affected Resource(s)

* google_v3.50.0

### Terraform Configuration Files

```
resource "google_compute_region_url_map" "test-url-map" {
  name            = "test-url-map"

  host_rule {
    hosts                 = ["myfqdn.here.com"]
    path_matcher          = "patchername"
  }

  path_matcher {
    name                  = "patchername"
    default_url_redirect {
      host_redirect    = "redirect.here.com"
      https_redirect   = true
      strip_query      = false
    }
  }
}

```

### Debug Output

```
main.tf in resource "google_compute_region_url_map" "test-url-map":
  95:   path_matcher {

The argument "default_service" is required, but no definition was found.
```

### Expected Behavior

Terraform plan to succeed and not requesting a default_service to be set when a default_url_redirect is set within a path_matcher block. As per the google documentation:

```
pathMatchers[].defaultService
...
Only one of defaultService, defaultUrlRedirect or defaultRouteAction.weightedBackendService must be set.
```

https://cloud.google.com/compute/docs/reference/rest/v1/regionUrlMaps

### Actual Behavior

```
main.tf in resource "google_compute_region_url_map" "test-url-map":
  95:   path_matcher {

The argument "default_service" is required, but no definition was found.
```

### Steps to Reproduce

```
resource "google_compute_region_url_map" "test-url-map" {
  name            = "test-url-map"

  host_rule {
    hosts                 = ["myfqdn.here.com"]
    path_matcher          = "patchername"
  }

  path_matcher {
    name                  = "patchername"
    default_url_redirect {
      host_redirect    = "redirect.here.com"
      https_redirect   = true
      strip_query      = false
    }
  }
}

```

1. `terraform plan`

### References

* #0000

**Comment by edwardmedia** [comment]
@Gilles95 do you notice the [default_service](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_url_map#default_service) field at the root? Let me know if that works

**Comment by Gilles95** [comment]
> @Gilles95 do you notice the [default_service](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_url_map#default_service) field at the root? Let me know if that works

@edwardmedia , thanks for following up. Yes I noticed it at the root, the root one is actually optional which is good, and we also have default_service within the path_matcher block, but that one is not optional and should be.
I am getting familiar with magic-module to PR it.

**Comment by edwardmedia** [comment]
@Gilles95 have you tried to put `default_service` at the root?

**Comment by Gilles95** [comment]
@edwardmedia, yes I did but the issue remains and the error point to the path_matcher line block.

**Comment by jbbqqf** [comment]
Hi! Triaging older `google_compute_region_url_map` reports — I think this one has been resolved.

In the current provider, `default_service` inside a `path_matchers` block is `Optional` and doesn't force any `ExactlyOneOf` group, so a path matcher that only sets `default_url_redirect` (or `default_route_action.weighted_backend_services`) is accepted by the schema. You can see it in [`google/services/compute/resource_compute_region_url_map.go`](https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/compute/resource_compute_region_url_map.go) — the nested `path_matchers.default_service` declaration is `Optional: true` with a `DiffSuppressFunc`, and no required/exactly-one-of constraint. The mmv1 source ([`mmv1/products/compute/RegionUrlMap.yaml#L404-L419`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/compute/RegionUrlMap.yaml#L404-L419)) has the older `exactly_one_of` block commented out.

So a config like the one you posted (path matcher with only `default_url_redirect`) should plan and apply cleanly today.

Would you mind closing this? If a more recent reproducer still trips a wrongly-required validation in `path_matchers`, drop the snippet here and I'll dig.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the schema and mmv1 source were verified manually against the current `terraform-provider-google` `main` branch.)
